src/qt-ui/main.cpp

```diff
@@ -0,0 +1,109 @@
+// system headers
+#include <chrono>
+#include <iterator>
+#include <iostream>
+#include <thread>
+
+// library headers
+#include <QCommandLineParser>
+#include <QFileDialog>
```

Comment:
**Caution** : File Dialog may break in some gtk based desktop environment. refer https://github.com/antony-jr/AppImageUpdater/issues/4

Reply:
Interesting. @probonopd please confirm.

Reply:
@TheAssassin Here are some other bugreports regarding QFileDialog , refer https://bugreports.qt.io/browse/QTBUG-33119 , https://bugreports.qt.io/browse/QTBUG-41416 , https://bugreports.qt.io/browse/QTBUG-63160 

This problem can be fixed if compiled against the latest version of Qt. The bug affects Qt v5.4.x and below. But this [bug](https://bugreports.qt.io/browse/QTBUG-63160) occurs on centos 7 even on the latest versions of Qt (v5.9.1).

Reply:
5.9.1 is not really up to date. Could you tell me from which version on the problem is fixed, please?

The Qt PPA specialist, Stephan Binner, told me he'd provide Qt 5.10.1 packages for Ubuntu trusty soon, so we can use that very soon.

Reply:
Lets first confirm the issue , I will do a test on ```ubuntu-mate-17.10-desktop-amd64.iso```.